src/javascript/class-scanner.ts

```diff
@@ -335,6 +335,199 @@ interface CustomElementDefinition {
+      const typeTag = jsdoc.getTag(jsdocAnn, 'type');
+      if (typeTag && typeTag.type) {
+        type = doctrine.type.stringify(typeTag.type);
+      }
+      description = getDescription(jsdocAnn);
+      privacy = getOrInferPrivacy(name, jsdocAnn);
+      readOnly = jsdoc.hasTag(jsdocAnn, 'readonly');
+    }
+
+    if (babel.isAssignmentExpression(node)) {
+      if (!type) {
+        const detectedType = closureType(node.right, sourceRange, this._document);
+        if (detectedType instanceof Warning) {
+          warnings.push(detectedType);
+          type = 'Function';
+        } else {
+          type = detectedType;
+        }
+      }
+    }
+
+    return {
+      name,
+      astNode: node,
+      type,
+      jsdoc: jsdocAnn,
+      sourceRange,
+      description,
+      privacy,
+      warnings,
+      readOnly,
+    };
+  }
+
+  private _createMethodFromExpression(
+      name: string, node: babel.FunctionExpression|babel.MemberExpression, jsdocAnn?: jsdoc.Annotation) {
+    let description;
+    let type;
+    let privacy: Privacy = 'public';
+    let ret;
+    const params = new Map<string, MethodParam>();
+
+    if (jsdocAnn) {
+      description = getDescription(jsdocAnn);
+      privacy = getOrInferPrivacy(name, jsdocAnn);
+      ret = getReturnFromAnnotation(jsdocAnn);
+      type = ret ? ret.type : undefined;
+
+      if (babel.isFunctionExpression(node)) {
+        (node.params || []).forEach((nodeParam) => {
+          const param = toMethodParam(nodeParam, jsdocAnn);
+          params.set(param.name, param);
+        });
+      } else {
+        for (const tag of (jsdocAnn.tags || [])) {
+          if (tag.title !== 'param' || !tag.name) {
+            continue;
+          }
+          let tagType;
+          let tagDescription;
+          if (tag.type) {
+            tagType = doctrine.type.stringify(tag.type);
+          }
+          if (tag.description) {
+            tagDescription = tag.description;
+          }
+          params.set(tag.name, {
+            name: tag.name,
+            type: tagType,
+            description: tagDescription
+          });
+        }
+      }
+    }
+
+    return {
+      name,
+      type,
+      description,
+      sourceRange: this._document.sourceRangeForNode(node)!,
+      warnings: [],
+      astNode: node,
+      jsdoc: jsdocAnn,
+      params: Array.from(params.values()),
+      return: ret,
+      privacy
+    };
+  }
+
+  private _createMemberFromMemberExpression(node: babel.ExpressionStatement) {
+    if (!babel.isMemberExpression(node.expression)) {
+      return;
+    }
+
+    const left = node.expression.object;
+
+    // we only want `something.prototype.member`
+    if (!babel.isMemberExpression(left) ||
+        !babel.isIdentifier(left.object) ||
+        !babel.isIdentifier(left.property) ||
+        !babel.isIdentifier(node.expression.property) ||
+        left.property.name !== 'prototype' ||
+        left.object.name !== this._name) {
+      return;
```

Comment:
is this the easiest way to do this? what about statics too? `Class.Foo;`. worth detecting them?

Reply:
This is the fastest way of doing it, though there's a more convenient way of doing it with `getIdentifierName` from ast-value.

Let's start with instance properties in this PR. Statics would be nice to get too though.